### ISSUE TITLE
Customized dataset training

### ISSUE THREAD

**Issue by drdr-jiang** [cvg/GeoCalib#20]
Hi, again requesting your guidance, I have customized the dataset for training with 4487 images and there are warnings generated during the training process, I saw in issues#17 you said “NAN detected in gradient clipping .skipping iteration.” this is a discard warning but I'm not sure what this means for my training! But I'm not sure if this is true for my training as well, because the gap between the weights file I get and the training data validation is very big, I'm asking for your help!

This is the training dataset:
![image](https://github.com/user-attachments/assets/1d26a813-8e36-4153-a45f-acb2c614bdc7)
This is warnning:
![2024-12-16_19-37](https://github.com/user-attachments/assets/039b3632-0d2c-4691-9455-14c596faae8b)
![2024-12-16_22-00](https://github.com/user-attachments/assets/1784e1ed-a982-431b-adf0-7dbad84f1aa6)
![2024-12-17_16-06](https://github.com/user-attachments/assets/dbd4a17e-fa2b-4566-a45b-09dcc88edc5e)
